crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1394,13 +1535,97 @@ pub async fn add_payment_method(
 
 #[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
 #[instrument(skip_all)]
-pub async fn add_payment_method(
-    _state: &routes::SessionState,
-    _req: api::PaymentMethodCreate,
-    _merchant_account: &domain::MerchantAccount,
-    _key_store: &domain::MerchantKeyStore,
-) -> errors::RouterResponse<api::PaymentMethodResponse> {
-    todo!()
+pub async fn vault_payment_method(
```

Comment:
Move these logic to where we are calling this function instead of having here, vault_payment_method_in_locker should be here and other logic should be caller side, we should avoid writing payment_method DB query in card.rs file.